add PhotonOS 3.0

bryceml opened the pull request:
### What does this PR do?
Adds PhotonOS 3.0 to the test suite

Note that `libsodium` and `libgit2-devel` are missing as they aren't in the photon os repos.

anitakrueger committed:
adding boto_elb config parameters

Paramters added to return for hosted_zone_name, hosted_zone_name_id and
vpc_id.

bobrik committed:
Do not render grains twice if sentry is disabled

We're seeing costly grains being rendered twice due to the sentry log
handler, which unconditionally renders grains for its own use. We're not
using Sentry with salt, so it seems quite wasteful.

cesan3 committed:
remove escaping of dbname in mysql.alter_db method.

https://github.com/saltstack/salt/issues/51559

cesan3 committed:
add issue file to changelog directory.

garethgreenaway commented:
https://github.com/saltstack/salt/pull/58991 should address the failing tests on Photon OS.

bryceml commented:
re-run full pr-photon3-py3-pytest

cesan3 committed:
fix: use __salt_system_encoding__ when retrieving certificate SHA1 str

- Also, add alternative method to obtain the SHA1 using
  crypto.load_certificate directly when not available when using x509 salt
  module.

anitakrueger committed:
adding test_get_elb_config function

meaksh committed:
Enhance logging when inotify beacon is missing pyinotify

meaksh committed:
Add changelog file

garethgreenaway commented:
re-run all

simonflood committed:
Update formulas.rst to reference LogBot archives

LogBot was decommissioned in mid-June 2021 so change the LogBot links to reference their Amazon S3-hosted archives instead.

Commit:
Fix #59033: salt-ssh error when targetting IPs or hostnames directly

Commit:
Add changelog

Commit:
Add ssh test when host is not in roster data

recursiveMake committed:
Add changelog entry for 60106 fix

noelmcloughlin committed:
docs(minion):: document master_type: str

Add missing "master_type: str" documentation as discussed at https://github.com/saltstack-formulas/salt-formula/pull/492#issuecomment-913149353

noelmcloughlin committed:
docs(minion): fix typo

dwoz committed:
Move resover stuff to tcp where it is useful

dwoz committed:
Refactor into transports and channels

waynew commented on tests/integration/modules/test_ssh.py:
```diff
@@ -128,6 +128,9 @@ def test_get_known_host_entries(self):
 
     @pytest.mark.slow_test
+    @pytest.mark.skipif(
+        salt.utils.platform.is_photonos() is True, reason="Skip on PhotonOS"
```
**question** can we say *why* we need to skip on photon here?

waynew replied:
(same for the rest of these)

garethgreenaway replied:
A lot of these will likely be removed once we figure out the SSH key thing.

waynew reviewed (changes requested):
Just some questions around skipping tests

meaksh committed:
Fix pylint issues

dwoz committed:
Clean up transports

dwoz committed:
Fix Serial references

dwoz committed:
Pickle PubServer

dwoz committed:
Fix pillar tests on macos

dwoz committed:
More test fixes

garethgreenaway committed:
Moving unit/utils/test_jinja.py to pytest.

garethgreenaway committed:
Fixing failing tests.  Fixing tmpdir paths.  Removing unneeded files.

garethgreenaway committed:
Fixing last couple of tests that were failing, removing more unneeded template files.

garethgreenaway committed:
Removing unneeded variables and imports.

Ch3LL commented:
Looks like some remaining tests and a merge conflict needing resolve are left on this one.

garethgreenaway committed:
Fixing fopens, converting the PosixPaths to strings to account for older versions of Python

dwoz committed:
Add pub server tests

garethgreenaway committed:
Fixing test_non_ascii tests.  Ensure we're writing out the file in the correct format.

garethgreenaway committed:
Converting from using tmpdir over to tmp_path.

dwoz committed:
Start to add base class defs

dwoz committed:
Remove raw opt from client send

dwoz committed:
Fix master_uri test

dwoz committed:
fix wart

garethgreenaway committed:
Moving test_pip_state to pytest.

garethgreenaway committed:
requested changes

garethgreenaway committed:
fixing skip fixture.

dwoz committed:
Clean up some docs and test fix

garethgreenaway committed:
Removing the fileserver_limit_traversal option from the default configuration as this is option is no longer used in code.

dwoz committed:
Address docs and hard coded strings

dwoz committed:
Fix bad import

Commit:
Fixes #59033: salt-ssh error when targetting IPs or hostnames directly

bdrung committed:
Fix 'BuildoutOnlineTestCase' has no attribute 'ppy'

The setup of `BuildoutOnlineTestCase.test_buildout_bootstrap` fails:

```
AttributeError: type object 'BuildoutOnlineTestCase' has no attribute 'ppy'
```

Signed-off-by: Benjamin Drung <benjamin.drung@ionos.com>

bdrung committed:
Mark test_buildout_bootstrap with pytest

Pytest runs `BuildoutOnlineTestCase.setUpClass` despite skipping all
tests. Mark `test_buildout_bootstrap` with pytest to avoid running the
`setUpClass`.

Signed-off-by: Benjamin Drung <benjamin.drung@ionos.com>

lukasraska committed:
Fix race condition in salt.utils.verify.verify_env

cbosdo committed:
virt: use ET node get() to fetch attributes

Since the XML "service" and "type" attributes may not be set on a VM
console, use the get() function as is returns a default None value

Commit:
Port file execution module tests to pytest

dwoz committed:
Not all transports have daemons

Commit:
fix failing non-windows tests

Commit:
fix more failing tests

Commit:
Replaced NamedTemporaryFile references with pytest tmp_path usage

Commit:
run pre-commit manually

Commit:
fix assert indentaion

Commit:
removed old tests

Commit:
remap filename_map.yml for new file exec module tests

dwoz committed:
Clear funcs cache channels

Commit:
Make directory in selinux tests

bdrung committed:
Fix spelling mistake of functionality

Signed-off-by: Benjamin Drung <benjamin.drung@ionos.com>

bdrung committed:
Fix spelling mistake of suppress

Signed-off-by: Benjamin Drung <benjamin.drung@ionos.com>

bdrung committed:
Fix spelling mistake of library

Signed-off-by: Benjamin Drung <benjamin.drung@ionos.com>

bdrung committed:
Fix spelling mistake of function

Signed-off-by: Benjamin Drung <benjamin.drung@ionos.com>

bdrung committed:
Fix spelling mistake of controlled

Signed-off-by: Benjamin Drung <benjamin.drung@ionos.com>

bdrung committed:
Fix spelling mistake of specified

Signed-off-by: Benjamin Drung <benjamin.drung@ionos.com>

bdrung committed:
Fix spelling mistake of available

Signed-off-by: Benjamin Drung <benjamin.drung@ionos.com>

bdrung committed:
Fix spelling mistake of arguments

Signed-off-by: Benjamin Drung <benjamin.drung@ionos.com>

bdrung committed:
Fix spelling mistake of "every time"

Signed-off-by: Benjamin Drung <benjamin.drung@ionos.com>

bdrung committed:
job_status: Convert datetime objects into formatted strings

`salt-call schedule.job_status` will return empty values for the
`datetime.datetime` objects. So convert the `datetime.datetime` objects
into formatted strings. Then they will be shown correctly:

```
$ sudo salt-call schedule.job_status __mine_interval
local:
    ----------
    _last_run:
        2021-11-01T12:36:57
    _next_fire_time:
        2021-11-01T13:36:57
    _next_scheduled_fire_time:
        2021-11-01T13:36:57
    _run_on_start:
        False
    _seconds:
        3600
    _splay:
        None
    enabled:
        True
    function:
        mine.update
    jid_include:
        True
    maxrunning:
        2
    minutes:
        60
    name:
        __mine_interval
    return_job:
        False
    run:
        True
    run_on_start:
        True
    splay:
        None
```

Signed-off-by: Benjamin Drung <benjamin.drung@ionos.com>

Commit:
run black on test_file_line.py

dwoz committed:
Add connect method to connect pub server channels

dwoz committed:
Fix pre-commit

dwoz committed:
add changelog

Commit:
Fix selinux test to account for pytest tmp_path fixture changes

dwoz committed:
Fix broken test

bdrung committed:
Fix misspelled import of M2Crypto in test_crypt.py

When running the tests from `tests/pytests/unit/utils/test_crypt.py`,
`test_random` fails when `M2Crypto` is installed:

```
$ python3 -m pytest -ra tests/pytests/unit/utils/test_crypt.py
[...]
FAILED tests/pytests/unit/utils/test_crypt.py::test_random -
AssertionError: assert CryptodomeRandom is None
```

This is caused by `salt.utils.crypt` importing `M2Crypto`, but the test
does not due to a spelling mistake. This was introduced in
74194770a8d50d3c8a2ccc2031de924435c4e495.

Signed-off-by: Benjamin Drung <benjamin.drung@ionos.com>

Commit:
Adjust rmtree references with only str args

dwoz committed:
Cleanup based on review comments

dwoz committed:
Fix ZeroMQ references

bdrung committed:
Fix test_ssh_kwargs to not access /etc/salt/pki

Running the SSH unit tests without root permission fails:

```
$ python3 -m pytest tests/pytests/unit/client/ssh/test_ssh.py
[...]
Traceback (most recent call last):
  File "salt/client/ssh/__init__.py", line 264, in __init__
    salt.client.ssh.shell.gen_key(priv)
  File "salt/client/ssh/shell.py", line 36, in gen_key
    os.makedirs(os.path.dirname(path))
  File "/usr/lib/python3.9/os.py", line 215, in makedirs
    makedirs(head, exist_ok=exist_ok)
  File "/usr/lib/python3.9/os.py", line 215, in makedirs
    makedirs(head, exist_ok=exist_ok)
  File "/usr/lib/python3.9/os.py", line 225, in makedirs
    mkdir(name, mode)
PermissionError: [Errno 13] Permission denied: '/etc/salt/pki'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "tests/pytests/unit/client/ssh/test_ssh.py", line 134, in test_ssh_kwargs
    ssh_obj = client._prep_ssh(**opts)
  File "salt/client/ssh/client.py", line 125, in _prep_ssh
    return salt.client.ssh.SSH(opts)
  File "salt/client/ssh/__init__.py", line 266, in __init__
    raise salt.exceptions.SaltClientError(
salt.exceptions.SaltClientError: salt-ssh could not be run because it could not generate keys.
```

Changing the accessed directories to temporary directories would be a solution:

```python
    client.opts["pki_dir"] = tmpdir.join("pki").strpath
    client.opts["cachedir"] = tmpdir.join("cache").strpath
```

But running the 38 tests takes 130 seconds on an Intel Core i7-8850H CPU @
2.60GHz. So mock all functions and methods that try to write to the pki or
cache directory. Then the tests will succeed in a few seconds.

Signed-off-by: Benjamin Drung <benjamin.drung@ionos.com>

Commit:
Pass str to fopen in test_file_line

dwoz committed:
Remove rabbitmq stuff for now

dwoz committed:
Cleanup and address PR comments

dwoz committed:
fix pre-commit

lukasraska committed:
Try/except for os.chown as well

Commit:
Add test and migrate spm test to pytest

bdrung committed:
Wrap long Python lines

Some source file includes a line length that is well beyond the normally
human made code line length.

Wrap long strings (use 20 bytes per line for the binary data). Also
blacken `salt/ext/vsan/vsanmgmtObjects.py` to make the code readable.

Signed-off-by: Benjamin Drung <benjamin.drung@ionos.com>

bdrung committed:
Wrap contributor list in release documentation

The release documentation files contain a list of contributors for the
release. This list is written in one line and can become very long,
which makes reading source file uneasy.

So wrap the list of contributions in the release documentation files.

Signed-off-by: Benjamin Drung <benjamin.drung@ionos.com>

bdrung committed:
Wrap long lines in release documentation

Signed-off-by: Benjamin Drung <benjamin.drung@ionos.com>

Commit:
Fix spm test for windows

Commit:
fix https://github.com/saltstack/salt/issues/60969

Commit:
Fixed bug with briding interface in debian #58195

bdrung committed:
Mention IP address if resolving FQDN fails

Resolving the FQDN for the IPv6 address failed in my virtual machine.
Salt produced an unhelpful error message:

```
[ERROR   ] Exception during resolving address: [Errno 2] Host name lookup failure
```

Change the log message to mention the IP address.

Signed-off-by: Benjamin Drung <benjamin.drung@ionos.com>

Commit:
work in progress while resolve issue of python3_32 usage by dnf and yum

cbosdo committed:
Fix salt-call event.send call with grains and pillar

Commit:
implement review requests for spm tests

Commit:
Implement review requests for spm changes

dwoz committed:
Add missing logic from refactor

twangboy committed:
Handle virtual accounts

Commit:
Add fix for https://github.com/saltstack/salt/issues/59554

twangboy committed:
Add tests, migrate some tests to pytest

twangboy committed:
Add changelog

dwoz committed:
Remove port info from channel doc strings

dwoz committed:
Fix param docs

dwoz committed:
Update publish_daemon docstring

dwoz committed:
Remove un-needed import

dwoz committed:
Use deepcopy for opts

dwoz committed:
Remove un-used import

dwoz committed:
Remove un-needed imports

dwoz committed:
Add deprecations for salt.transport.(client,server)

dwoz committed:
Document deprecation in docstring

twangboy committed:
Fix some lint

garethgreenaway committed:
Addressing a few issues when having keep_symlinks set to True with file.recurse.  Also allow symlinks that are outside the salt fileserver root to be discoverable as symlinks when fileserver_followsymlinks is set to False.

garethgreenaway committed:
adding changelog.

dwoz committed:
Call parent class no-op __init__

Commit:
Add changelog

garethgreenaway committed:
Fixing failing tests related to lack of support for PosixPath

Commit:
Prevent get_tops from performing a Set operation on a List

dwoz committed:
Use salt.channel.client instead of salt.transport.client

dwoz committed:
fix unit tests

krionbsd committed:
Move MacOS tests to pytest

dwoz committed:
Fix more tests

dwoz committed:
Clean up salt.transport.(client,server) references

dwoz committed:
Fix reference to salt.transport.client

dwoz committed:
Test tcp test fix

garethgreenaway committed:
Updating tests to work on FBSD and OS X.

garethgreenaway committed:
skip tests on Windows since it does not support symlinks

dwoz committed:
Try tcp test fix

dwoz committed:
Add message logging

garethgreenaway committed:
update rabbitmq tests to include version 3.9

bdrung committed:
docs: Support building documentation locally

Building the documentation locally without Internet access will fail,
because sphinx will try to download the intersphinx inventory from
http://docs.python.org/3/objects.inv.

Support building the documentation locally without Internet access by
searching for locally installed intersphinx inventory first.

Bug-Debian: https://bugs.debian.org/755026
Signed-off-by: Benjamin Drung <benjamin.drung@ionos.com>

dwoz committed:
test tcp

Commit:
Add prefixlen for IPv6 addresses

Commit:
Add multimaster failover tests

Commit:
Fix 'Path' use in 'copyfile'

Commit:
- Resolves "[BUG] regressions in mysql_cache module #61222"

Commit:
- fix up tests to match

Commit:
Work-around using LIBPATH for 32-bit python in new DNF and YUM support

Commit:
Bind masters to same interface

Commit:
Revert back to two separate interfaces

MKLeb committed:
Fix typo

Co-authored-by: Megan Wilhite <mwilhite@vmware.com>

waynew committed:
Add iptables doc example

Fixes #60880

twangboy committed:
Use `get` to account for missing key

Commit:
Use caplog fixture

fix test_pki comment

Commit:
Working tests for install

waynew committed:
Delete 60880.fixed

waynew committed:
Add tests for ansible roster virtual

waynew committed:
Fix ansible roster

Docs failed to mention `--roster ansible` and there was an issue with
the virtual function when ansible wasn't actually installed.

waynew committed:
Add changelog, complete docs

Commit:
Oop, forgot to whitelist for Windows

Commit:
Complete intial tests for AIX yum and dnf support

dwoz committed:
Test fix

dwoz committed:
Fix pre-commit

eliasp committed:
doc/win_repo: clarify when cache_dir/cache_file are actually effective

binocvlar committed:
Add testcase written by @waynew

dwoz committed:
Test tcp message client refactor fix

dwoz committed:
fix pre-commit

garethgreenaway committed:
Merge freeze into master (#61310)

* Redirect imports of ``salt.ext.six`` to ``six``

Fixes #60966

* Latest changelog update for 3004

* Handle signals and properly exit, instead of raising exceptions.

This was introduced in 26fcda50740c92fe3e4a5ca50a0d83227009942a

Fixes #60391
Fixes #60963

* Add test for #61003

* Fix #61003

Restored the previously shifted check for version_to_remove in
old[target]. This had been extracted along with the correctly extracted
double pkg_params[target] lookup, but that lost the `target in old`
guard.

Putting the check back here prevents KeyError when looking for a
non-existent target in `old`.

* Handle various architecture formats in aptpkg module

* Write file even if does not exist

* only run test on debian based platforms

* remove extra space for arch

* convert pathlib to string for pkgrepo test

* Use temporary files first then copy to sources files

* fixes saltstack/salt#59182 fix handling of duplicate keys in rest_cherrypy data

* added changelog

* remove log messages to prevent leaks of sensitive info

* Reverting changes in PR #60150. Updating installed and removed functions to return changes when test=True.

* Adding changelog.

* Add a test and fix for extra-filerefs

* Do not break master_tops for minion with version lower to 3003

* Add changelog file

* Add extra comment to clarify discussion

* Update changelog file

* Add deprecated changelog

* Assert that the command didn't finish

Refs https://github.com/saltstack/salt/pull/60972

* Always restore signals, even when exceptions occur

* Reset signal handlers before starting the process

* Make sure that the `ProcessManager` doesn't always ignore signals

* Provide valid default value for bootstrap_delay

* Update changelog for 3004

* Update changelog and release notes for 3004

* Add PR 61020 to changelog

* Update release notes index file and create 3005 file

* Update docs

* bad merge from master, redoing the PR for adding the Deltaproxy documentation.

* Adding doc/ref/configuration/delta_proxy.rst.

* cleaning up this paragraph, adding in network administrators in addition to system administrators.

Co-authored-by: Pedro Algarvio <pedro@algarvio.me>
Co-authored-by: ScriptAutomate <derek@icanteven.io>
Co-authored-by: Wayne Werner <wwerner@vmware.com>
Co-authored-by: Megan Wilhite <mwilhite@vmware.com>
Co-authored-by: nicholasmhughes <nicholasmhughes@gmail.com>
Co-authored-by: Daniel A. Wozniak <dwozniak@saltstack.com>
Co-authored-by: Pablo Suárez Hernández <psuarezhernandez@suse.com>
Co-authored-by: Alyssa Rock <arock@saltstack.com>
Co-authored-by: Twangboy <shane.d.lee@gmail.com>

Commit:
- Normalize and update all run_query() calls to use kwargs

Commit:
- format to make the linter happy

Commit:
- remove kwargs for positional args in run_query()

Commit:
Updates from checking changes on AIX 7.2 TL4

Commit:
- Dang it, why all the dangling commas at the end of the expressions?

Commit:
- ARGH!  "Fixing" code that's NOT BROKEN is a huge pain

Commit:
Add regular multimaster tests

dwoz committed:
Fix broken unit test

Commit:
- This is LITERALLY pushing me stop submitting code to salt....

Commit:
precommit

krionbsd committed:
Move tests to pytest (#61354)

Commit:
Updated to allow for old Python 2 based yum on AIX

Commit:
Added changelog entry

waynew committed:
Fix typo

Commit:
fix failing test? maybe?

Commit:
Fixed doc generation error

dwoz committed:
Clean up doc strings

dwoz committed:
Enable topics for tcp transport

Commit:
Updated to remove use of f-strings, and review comments

binocvlar committed:
Mutate set rather than constructing a new one

As favoured by @waynew

dwoz committed:
Deltaproxy does not work with subscriptions

ari committed:
Improve documentation for FreeBSD pkg

https://github.com/saltstack/salt/issues/23746 is never likely to be "fixed" and its not even clear it should be, so just simplify the warning so that it makes sense.

Package list is already supported in pkgng.

Commit:
Bump duration time for windwos for test_retry_option_success

Commit:
fix failing `test_store`

sblaisot committed:
Handle new MariaDB SLAVE MONITOR grant

twangboy committed:
Use net instead of sc to restart minion service

meaksh committed:
Align enhanced logging accross beacons

meaksh committed:
Fix changelog file

meaksh committed:
Fix failing unit tests

bdrung committed:
Fix test cases with PermissionError on /var/cache/salt

When running the test cases without root permission, some test cases fail:

```
$ python3 -m pytest -ra tests/pytests/unit/state/test_state_compiler.py tests/pytests/unit/test_minion.py
[...]
FAILED tests/pytests/unit/state/test_state_compiler.py::test_render_requisite_require_disabled - PermissionError: [Errno 13] Permission denied: '/var/cache/salt'
FAILED tests/pytests/unit/state/test_state_compiler.py::test_render_requisite_require_in_disabled - PermissionError: [Errno 13] Permission denied: '/var/cache/salt'
FAILED tests/pytests/unit/test_minion.py::test_minion_module_refresh - PermissionError: [Errno 13] Permission denied: '/var/cache/salt'
FAILED tests/pytests/unit/test_minion.py::test_minion_module_refresh_beacons_refresh - PermissionError: [Errno 13] Permission denied: '/var/cache/salt'
```

Fix these test cases by using a temporary directory as cache directory.

Signed-off-by: Benjamin Drung <benjamin.drung@ionos.com>

krionbsd committed:
Update pytest to 6.2.5 for python 3.10 (#61295)

* Update pytest to 6.2.5 for python 3.10

* bump junos-eznc to 2.6.0

* bump setproctitle version and fix integration tests

* Skip tests for python 3.10

* Fix test with py3.10

* bump jinja2 version and dependencies

* Fix test with py3.10

* Skip tests for python 3.10 and new moto version

* Use tempfile library to test perms on directories

* Maintain static requirements changes, revert `requirements/base.txt`

Signed-off-by: Pedro Algarvio <palgarvio@vmware.com>

Co-authored-by: Megan Wilhite <mwilhite@vmware.com>
Co-authored-by: Pedro Algarvio <palgarvio@vmware.com>

dwoz committed:
Address PR comments

nicholasmhughes committed:
add etag support for file.managed web sources

fkantelberg committed:
Fix the check in the hetzner cloud show_instance function to be an action.

twangboy committed:
Add changelog

Commit:
fix multimaster tests failing

Commit:
format long line

nicholasmhughes committed:
refactor checking for web source hash verification

Commit:
fix multimaster tests, but make it the sequel

fkantelberg committed:
Update salt/cloud/clouds/hetzner.py

Co-authored-by: Caleb Beard <53276404+MKLeb@users.noreply.github.com>

Commit:
fix tcp tests

garethgreenaway committed:
Adding the ability to add, delete, purge, and modify Salt scheduler jobs when the Salt minion is not running.

Commit:
bump lxml to >=4.6.5

nicholasmhughes committed:
adding functional tests for use_etag parameter in file.managed state

Commit:
failover fixes, fingers crossed

Commit:
fix wait time

Commit:
wait time fix v2

nicholasmhughes committed:
patch TestRequestHandler with directory feature added after python 3.6

nicholasmhughes committed:
handling different python http modules is fun

nicholasmhughes committed:
python 3.5 doesnt like to os.path.join posixpath objects

nicholasmhughes committed:
fixes #61270 adds etag support for file.managed web sources

meaksh committed:
Fix function to check during smartos_vmadm beacon virtual

garethgreenaway committed:
renaming changelog.

twangboy committed:
Update the macOS installer and docs

Update the welcome and conclusion pages of the macOS installer
Bring some of the information in the welcome to the conclusion page
Add clarification about the salt-config cli utility
Add docs to the installation page for macOS for the installer
Add docs to the installation page for macOS for the salt-config tool

twangboy committed:
Add changelog

twangboy committed:
Edit RST files on a mac so that line endings and fonts are correct

nicholasmhughes committed:
fixes #61440 and removes case sensitivity of Etag header

dwoz committed:
Fix docstrings

Commit:
Uses only command's stdout to populate lvm grain

krionbsd committed:
Change to https:// URL

Commit:
fix wait times, but make it all of them

twangboy committed:
Expand environment var from registry

twangboy committed:
Add changelog

dwoz committed:
Skip new deltaproxy tests on tcp transport

twangboy committed:
Add tests

garethgreenaway committed:
fixing failing test

twangboy committed:
Update port install info, remove errata, remove setting maxfiles

twangboy committed:
Skip windows specific tests on non-windows systems

garethgreenaway committed:
Fixing failing test.  Order of OrderedDict not guaranteed on older Python versions.

garethgreenaway committed:
Adding the changelog back in.

dwoz committed:
Add NotImplimentedError to stubs

dwoz committed:
Fix lint error

dwoz committed:
Fix typo

dwoz committed:
Fix base class init methods

priyank-R committed:
Updated renderer docs mentioning gpg as a text renderer to both, text and data renderer

Commit:
change return checking strategy

garethgreenaway committed:
Fixing tests to use the correct paths for the schedule configuration file.

waynew committed:
Fix for pre-commit

garethgreenaway committed:
Fixing test_purge failures.

garethgreenaway committed:
Removing unnecessary fixture.

Commit:
move git exec module tests to pytest

Commit:
move cp exec module tests to pytest

Commit:
move chocolatey exec module tests to pytest

garethgreenaway committed:
Adding changelog.

mchugh19 committed:
attempt at fixing 25721 on master (#55720)

* master port

* Add `timeout_decorator` as a testing dependency

* Switch to use pytest-timeout instead

* fix pre-commit

* Add changelog

Co-authored-by: Pedro Algarvio <pedro@algarvio.me>
Co-authored-by: Megan Wilhite <megan.wilhite@gmail.com>
Co-authored-by: Megan Wilhite <mwilhite@vmware.com>

dwoz committed:
3004 has been released

Commit:
fix global variables

jfindlay committed:
Test that sentry logger does not load grains/modules

Reinforce sentry log handler not wasting time reloading grains/execution
modules with unit tests.

priyank-R committed:
Updates file.absent documentation to include details on how to clear directory contents

garethgreenaway committed:
Removing fileserver_limit_traversal from all configuration files and documentation.

Commit:
Fix line character limit in file docs

garethgreenaway committed:
[merge jam] Master port 49261 - consul modules (#58101)

* add consul states and acl function present/absent

* add consul to states doc index

* refact/fix consul states

* fix doc, fix states

* fix name parameter for acl_changes

* fixing pylint errors

* small changes after review by @rallytime

* fix header count

* Update consul.py

* fix acl_exists description, fix when both id and name are missing

* Adding some tests for consul module and consul state module.  Some additional fixes in the consul module.

* Fixing tests.

* Fixing failing tests on Windows.

* Adding changelog.

* Adding some tests for consul module and consul state module.  Some additional fixes in the consul module.

* moving tests to pytest.

* manual black changes.

* One more manual black change.

* fixing formatting.  Adding versionadded for state module.

Co-authored-by: Rémi Jouannet <remi.jouannet@outscale.com>
Co-authored-by: Mike Place <mp@saltstack.com>
Co-authored-by: Daniel Wozniak <dwozniak@saltstack.com>
Co-authored-by: Wayne Werner <wwerner@vmware.com>

ggiesen committed:
VMware expanded virtual hardware features (#60739)

* fixes #56144 to enable hotadd profile support

* updates for black

* complying with new tests that were recently added

* Add suppor for cpuHotRemoveEnabled, nestedHVEnabled, vPMCEnabled

* Update docs for cpuHotRemoveEnabled, nestedHVEnabled, vPMCEnabled

* Add checks for platform and primary_ip and print errors if they are missing when proxy_return is set

* Add changelog

* Add changelog

* Roll back commit and properly add files

* Revert "Roll back commit and properly add files"

This reverts commit 59fd715c8ca2a563412372c31f83c0df4ffc15f8.

* Revert "Add changelog"

This reverts commit 036fc35ce9c673a58f3b0d95d6983fea39b5fd82.

* Revert "Add checks for platform and primary_ip and print errors if they are missing when proxy_return is set"

This reverts commit 090aa3ab3689121c0e34aac33734bbc181dac7ab.

Co-authored-by: nicholasmhughes <nicholasmhughes@gmail.com>
Co-authored-by: Dmitry Kuzmenko <dmitry.kuzmenko@dsr-corporation.com>
Co-authored-by: Daniel Wozniak <dwozniak@saltstack.com>
Co-authored-by: Sage the Rage <36676171+sagetherage@users.noreply.github.com>

garethgreenaway commented:
This one is super old and outdated, closing it out in favor of  #61500
